packages/protobuf-test/src/is-message.test.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { describe, expect, test } from "@jest/globals";
+import { User as TS_User } from "./gen/ts/extra/example_pb.js";
+import { User as JS_User } from "./gen/js/extra/example_pb.js";
+import { isMessage } from "@bufbuild/protobuf";
+
+describe("isMessage", () => {
```

Comment:
We're missing a test in the type system. isMessage is a type guard, and we should test whether it works.

This should cover the two use cases:

```ts
const user: unknown = new TS_User();
if (isMessage(user)) {
  expect(user.toJsonString).toBeDefined();
}
if (isMessage(user, TS_User)) {
  expect(user.firstName).toBeDefined();
}
```